### PR TITLE
Re-add dev to GH actions

### PR DESCRIPTION
**Pull request by illusional**

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
   push:
     branches:
       - main
+      - dev
```
Comment (lgruen) [inline .github/workflows/unit-tests.yml:9]: Would it be helpful to run these on any branch during development?

Comment (illusional) [reply .github/workflows/unit-tests.yml:9]: This runs on pull requests already, so I interpret this as another safety run essentially.

```
  pull_request:
    types: [opened, synchronize, reopened]
```

**Comment by illusional** [comment]
One of the tests failed with: 

> FAIL: test_dashboard_page_data (seqr.views.apis.dashboard_api_tests.LocalDashboardPageTest)

I think this got disabled in the next commit, but just highlighting it as untouched by this PR.

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (lgruen) [reply .github/workflows/unit-tests.yml:9]: Got it, thanks!

**Review by lgruen — APPROVED** [review]

**Pull request merged**